src/frontends/tensorflow/src/decoder_argdef.cpp

```diff
@@ -0,0 +1,86 @@
+        {::tensorflow::DataType::DT_BOOL, ov::element::boolean},
+        {::tensorflow::DataType::DT_INT16, ov::element::i16},
+        {::tensorflow::DataType::DT_INT32, ov::element::i32},
+        {::tensorflow::DataType::DT_INT64, ov::element::i64},
+        {::tensorflow::DataType::DT_HALF, ov::element::f16},
+        {::tensorflow::DataType::DT_FLOAT, ov::element::f32},
+        {::tensorflow::DataType::DT_DOUBLE, ov::element::f64},
+        {::tensorflow::DataType::DT_UINT8, ov::element::u8},
+        {::tensorflow::DataType::DT_INT8, ov::element::i8},
+        {::tensorflow::DataType::DT_BFLOAT16, ov::element::bf16}};
+    return type_map;
+}
+}  // namespace
+
+size_t DecoderArgDef::get_input_size() const {
+    FRONT_END_GENERAL_CHECK(m_op_type == "input_arg" || m_op_type == "output_arg",
+                            "[TensorFlow Frontend] Internal error: Incorrect use of DecoderArgDef class.");
+    if (m_op_type == "input_arg") {
+        return 0;
+    } else {
+        return 1;
+    }
+}
+
+const std::string& DecoderArgDef::get_op_type() const {
+    FRONT_END_GENERAL_CHECK(m_op_type == "input_arg" || m_op_type == "output_arg",
+                            "[TensorFlow Frontend] Internal error: Incorrect use of DecoderArgDef class.");
+    return m_op_type;
+}
+
+const std::string& DecoderArgDef::get_op_name() const {
+    return m_arg_def->name();
+}
+
+void DecoderArgDef::get_input_node(size_t input_port_idx,
+                                   std::string& producer_name,
+                                   size_t& producer_output_port_index) const {
+    FRONT_END_GENERAL_CHECK(m_op_type == "output_arg",
+                            "[TensorFlow Frontend] Internal error: get_input_node is supported only for output_arg.");
+    auto first_colon = m_producer_name.find_first_of(":");
+    auto last_colon = m_producer_name.find_last_of(":");
+    if (first_colon != std::string::npos && last_colon != std::string::npos) {
+        producer_name = m_producer_name.substr(0, first_colon);
+        auto port_id = m_producer_name.substr(last_colon + 1);
+        FRONT_END_GENERAL_CHECK(!port_id.empty() && std::all_of(port_id.begin(), port_id.end(), ::isdigit),
+                                "Port id is not specified or not a number. Value: ",
+                                port_id);
+        producer_output_port_index = std::stoi(port_id);
+        return;
+    }
+    producer_name = m_producer_name;
+    producer_output_port_index = 0;
+}
+
+ov::Any DecoderArgDef::get_attribute(const std::string& name) const {
+    FRONT_END_GENERAL_CHECK(name == "type",
+                            "[TensorFlow Frontend] Internal error: DecoderArgDef supports only `type` attribute.");
+    if (TYPE_MAP().count(m_arg_def->type())) {
+        return TYPE_MAP().at(m_arg_def->type());
+    } else {
+        // for all unsupported types return undefined type
+        return ov::element::undefined;
```

Comment:
Why undefined? I think better would be `dynamic`

Reply:
No, `dynamic` means a type of dynamic size, etc. This case is more close when we don't know of the type nothing.

Reply:
`dynamic` means that we don't know the type, it can serialize and deserialize fine. `dynamic` can be merged with static type. While `undefined` can't be deserialized, and can't be merged with any static type. I suggest using `dynamic` unless you have strong reason to not use it.

Reply:
We have already use such approach with `undefined` type for `NodeDef` decoder. Also, we have translators for string `Const` nodes.

Also, I would rather to know with which type we have a problem with and serialization can highlight this.